### DISCUSSION TITLE
Merge Queues: Behavior when queued PRs are updated

### DISCUSSION THREAD

**Discussion by nacx** [community/community#31588] [Pull Requests]
First of all, thanks for adding this great feature! 

In the last month, I experienced some unexpected behavior when using Merge Queues with pull requests that were updated with new commits. When a PR that passes all checks and is approved is moved to the merge queue:

* If you rebase the PR or add some additional commits, the code in the merge queue is not evicted but it is not updated either, which no longer reflects the current status of the PR.
* The queue continues, and if everything passes, it merges the changes (the ones in the queue branch, so the new commits in the PR are not included).
* The PR is marked as merged even when the last commits were not added to the queued build. Unless you check explicitly you don't know that the merged PR is actually just partially merged.

**Comment by azarudeena** [comment]
The moment when the PR enters is to the merge queue following changes won't be reflected in the PR. Even there is a warning message for that.
